docs/design/README.md

```diff
@@ -539,15 +539,21 @@ the array may be deduced, it may be omitted, as in:
 
 ```carbon
 var i: i32 = 1;
+// [i32;N] [i32;N] is an array representing N values, the numerical type of each value is the data type of i32, the size of the array is determined, and new elements cannot be added, nor can the elements of the array be reduced.
+   var a:[i32;5] = (2,4,5,7,9);
 // `[i32;]` equivalent to `[i32; 3]` here.
-var a: [i32;] = (i, i, i);
+var a: [i32;] = (i,j,k);
 ```
 
 Elements of an array may be accessed using square brackets (`[`...`]`), as in
-`a[i]`:
+`a[0]`:
 
 ```carbon
-a[i] = 2;
+// a[i] = 2;error,The subscript value of a[i] is a definite number, not i
```

Comment:
Making this an error would be a major design change, and would drastically limit the usefulness of arrays. Are you intending to propose such a change? If so, why?

Reply:
This wrong design is not easy to understand. It is not very friendly to developers. It is difficult to understand such a wrong design. Combined with other languages such as rust and C++, there is no such wrong design. Besides, such a design , when running, it will bring uncertainty to the memory.

Reply:
> Making this an error would be a major design change, and would drastically limit the usefulness of arrays. Are you intending to propose such a change? If so, why?


This wrong design is not easy to understand. It is not very friendly to developers. It is difficult to understand such a wrong design. Combined with other languages such as rust and C++, there is no such wrong design. Besides, such a design , when running, it will bring uncertainty to the memory.

Reply:
Using a variable as an array index is definitely not an error in C++ ([example](https://godbolt.org/z/949j91xrq)) or Rust ([example](https://godbolt.org/z/q1PhKooGe)), so I think we must be miscommunicating somehow.